doc/specs/#2325 - Default Profile Settings.md

```diff
@@ -0,0 +1,314 @@
+            "fontFace": "Cascadia Code",
+            "fontSize": 10
+        },
+        {
+            "guid": "{61c54bbd-c2c6-5271-96e7-009a87ff44bf}",
+            "inheritFrom": "{11111111-1111-1111-1111-111111111111}",
+            "name": "Windows PowerShell",
+            "commandline": "powershell.exe",
+            "hidden": false
+        },
+        {
+            "guid": "{0caa0dad-35be-5f56-a8ff-afceeeaa6101}",
+            "inheritFrom": "{11111111-1111-1111-1111-111111111111}",
+            "name": "cmd",
+            "commandline": "cmd.exe",
+            "hidden": false
+        },
+        {
+            "guid": "{0caa0dad-ffff-5f56-a8ff-afceeeaa6101}",
+            "inheritFrom": "{0caa0dad-35be-5f56-a8ff-afceeeaa6101}",
+            "name": "This is another CMD",
+            "commandline": "cmd.exe /c myCoolScript.bat",
+            "hidden": false
+        }
+    ],
+    "schemes": [],
+    "keybindings": []
+}
+```
+
+#### Benefits
+
+##### Matches the existing settings model without major refactoring
+Simply adding a new property to `Profile` would not majorly alter the structure
+of the file.
+
+##### Property name is unique
+`inheritFrom` is very unique relative to other keys we already have.
+
+##### Powerful
+This lets the user have potentially many layers of settings grouping. hese
+layers would let the user seperate out common settings however they like,
+without forcing them to a single "default" profile. They could potentially have
+many "default" profiles, e.g.
+* one that's used for all their WSL profiles, with `startingDirectory` set to
+  `~` and `fontFace` set to "Ubuntu Mono"
+* One that's used for all their powershell profiles
+
+etc.
+
+#### Concerns
+
+##### GUIDs are not human friendly
+
+Using the guid in the `inheritFrom` field is the only way to be sure we're
+uniquely identifying profiles. However, guids are notoriously un-friendly. The
+above example manually uses `"{11111111-1111-1111-1111-111111111111}"` as the
+guid of the "default" profile, but inheriting from other profiles with "real"
+GUIDs would be less understandable. Consider the "This is another CMD" case,
+where it's inheriting from the "cmd" profile. That `"inheritFrom"` value does
+not mean at a quick glance "cmd".
+
+##### We have to make sure that there are no cycles as we're layering
+
+This is mostly a technical challenge, but this does make the implementation a
+bit trickier.
+
+##### How does this work with the settings UI?
+
+When the user edits settings for a profile with the UI, do we only place the
+changes in the top-most profile?
+
+How do we communicate in the UI that a profile is inheriting settings from other
+profiles?
+
+##### Harder to mentally parse
+Maybe not as easy to mentally picture how one profile inherits from another. The
+user would probably need to manually build the tree of profile inheritance in
+their own head to understand how a profile gets its settings.
+
+## Conclusions

```

Comment:
```
### Proposal 5: Demote profiles _and_ profile settings, the VS Code model
```

VS Code does this:

```jsonc
"editor.fontFace": "Cascadia Code",
// ...
"[cpp]": {
    "editor.fontFace": "Inconsolata DZ" // C++ looks better like this
}
```













having written it up, i hate it.

Reply:
Yea I hate this too. I'm going to just resolve this thread since it's got two vetos.